server/trillian_log_server.go

```diff
@@ -222,7 +222,7 @@ func (t *TrillianLogServer) commitAndLog(tx storage.LogTX, op string) error {
 }
 
 func protoToLeaf(proto *trillian.LeafProto) trillian.LogLeaf {
-	return trillian.LogLeaf{Leaf: trillian.Leaf{LeafHash: proto.LeafHash, LeafValue: proto.LeafData, ExtraData: proto.ExtraData}}
+	return trillian.LogLeaf{SequenceNumber: proto.LeafIndex, Leaf: trillian.Leaf{LeafHash: proto.LeafHash, LeafValue: proto.LeafData, ExtraData: proto.ExtraData}}
```

Comment:
looks like it does - should you be copying SequenceNumber across from proto to LogLeaf too?